### DISCUSSION TITLE
CSharpClientGenerator : add config flag to use ConfigureAwait(false) optionally

### DISCUSSION THREAD

**Discussion by dpradell-dev** [RicoSuter/NSwag#4328] [Ideas]
Hi @RicoSuter,

First of all, thanks for your amazing work with NSwag. I'm using the `CSharpClientGenerator` to autogenerate a Unity SDK and so far it's working great in most platforms :)

The only issue I have is that my implementation doesn't require `ConfigureAwait(false)` as it's using UniTask. Would it be possible to add a **config flag** so we can set it to `false` if we don't want to include `ConfigureAwait(false)` at the end of every request?

Thanks again,

David

**Comment by StuffOfInterest** [comment]
`ConfigureAwait(false)` caused an issue for me due to it losing the HttpContext when using the sync version of a method in some .NET Framework code.  I was able to work around the issue, but being able to not include that would be helpful.  In many cases, I believe it isn't even recommended anymore.
